src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingService.kt

```diff
@@ -893,6 +901,45 @@ class BookingService(
     booking.departureDate = expectedDepartureDate
     updateBooking(booking)
 
+    if (booking.premises is TemporaryAccommodationPremisesEntity) {
+      val domainEventId = UUID.randomUUID()
+
+      val application = booking.application as? TemporaryAccommodationApplicationEntity
+
+      cas3DomainEventService.savePersonArrivedEvent(
```

Comment:
Now we have everything laid out on how to build this domain event, I wonder if we know enough to abstract this knowledge and responsibility out of the booking service and push it down a layer. This new bit of indirection feels worth it and it'd make/keep the booking class simpler.

Something like `cas3DomainEventService.savePersonArrivedEvent(booking)` ?

Reply:
I've introduced a `DomainEventBuilder` class that can be used to obtain an instance of an event.